src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate.cs

```diff
@@ -262,32 +262,42 @@ public virtual String ToString(bool fVerbose)
             StringBuilder sb = new StringBuilder();
 
             // Subject
-            sb.Append("[Subject]" + Environment.NewLine + "  ");
-            sb.Append(this.Subject);
+            sb.AppendLine("[Subject]");
+            sb.Append("  ");
+            sb.AppendLine(this.Subject);
 
             // Issuer
-            sb.Append(Environment.NewLine + Environment.NewLine + "[Issuer]" + Environment.NewLine + "  ");
-            sb.Append(this.Issuer);
+            sb.AppendLine();
+            sb.AppendLine("[Issuer]");
+            sb.Append("  ");
+            sb.AppendLine(this.Issuer);
 
             // Serial Number
-            sb.Append(Environment.NewLine + Environment.NewLine + "[Serial Number]" + Environment.NewLine + "  ");
+            sb.AppendLine();
+            sb.AppendLine("[Serial Number]");
+            sb.Append("  ");
             byte[] serialNumber = this.GetSerialNumber();
             Array.Reverse(serialNumber);
-            sb.Append(serialNumber.ToHexStringUpper());
+            sb.AppendLine(serialNumber.ToHexStringUpper());
```

Comment:
It seems to me like ToHexStringUpper should return char[] instead of string and then the char[] should be appended here.